client/landing/gutenboarding/index.tsx

```diff
@@ -30,7 +32,14 @@ window.AppBoot = () => {
 
 		ReactDom.render(
 			<BrowserRouter basename="gutenboarding">
-				<Gutenboard />
+				<Switch>
+					<Route exact path={ path }>
+						<Gutenboard />
```

Comment:
Here I would redirect to a path without `/LANG`. I see a common pattern used in current signup flow. You can access any of the steps with `/LANG` and you'll set that language and the LANG param will be immediately removed from URL.
Also you'll be redirected to the first step if you tried to jump in the middle of the flow but with the language you asked. Eg: https://wordpress.com/start/plans/fr

Reply:
> Here I would save the provided language to `USER_STORE` and then redirect to a path without `/LANG`. I see a common pattern used in current signup flow. You can access any of the steps with `/LANG` and you'll set that language and the LANG param will be immediately removed from URL.

I'm a bit rusty with regards to the ins and outs of routing, but the way it's done in the current signup flow isn't without its problems. I think we should consider arguments for both retaining and dropping the language route fragment; the fact that it's dropped in the current signup flow isn't convincing enough on its own :grimacing:

Reply:
Retaining the language route fragment might be useful when sharing the URL with other users or a second device. But we don't currently store much data in the URL itself so any Gutenboarding progress would be lost when accessed in some other browser. 

On the other side, the added complexity to the routing can generate a series of new bugs. Try for example going to [/gutenboarding/fr](https://hash-0bb946e260062e057d458e0bef81e68c64d7b7b5.calypso.live/gutenboarding/fr) and press next. Or go to [/gutenboarding/pages/fr](https://hash-0bb946e260062e057d458e0bef81e68c64d7b7b5.calypso.live/gutenboarding/pages/fr) and press anywhere.

Reply:
I don't see benefit to moving a piece of state (language) that's explicit in the URLs into hidden application state and applying all the redirection logic. When we don't know a language, let's keep an explicit one on the URL throughout the flow.

> On the other side, the added complexity to the routing can generate a series of new bugs. Try for example going to [/gutenboarding/fr](https://hash-0bb946e260062e057d458e0bef81e68c64d7b7b5.calypso.live/gutenboarding/fr) and press next. Or go to [/gutenboarding/pages/fr](https://hash-0bb946e260062e057d458e0bef81e68c64d7b7b5.calypso.live/gutenboarding/pages/fr) and press anywhere.

The bugs should be ironed out, this wasn't ready for testing in these examples. Redirecting and removing language slugs introduces its own complexities.